LanguageFeatures/Parts-with-imports/scope_A05_t02_part2.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let P be a prefix scope containing all the import prefixes
+/// declared by the current file. The parent scope of P is I.
+/// ...
+/// - If an import is `deferred`, its Pname is a deferred scope which
+///   has an extra `loadLibrary` member added, as usual, and the import has an
+///   implicit `hide  loadLibrary` modifier.
+///
+/// @description Check that if an import is deferred an extra `loadLibrary`
+/// member is added and it is a runtime error to access any of its members
+/// before `loadLibrary()` completes successfully. Test the case when
+/// `loadLibrary()` is called from another part file.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=enhanced-parts
+
+part of 'scope_A05_t02_part1.dart';
+
+testPart2() async {
+  // From scope_lib1.dart
+  Expect.throws(() {print(l1.libVar);});
```

Comment:
I suspect this could fail with some backends (perhaps DDC?) because they do not implement deferred libraries, in a sense, they just load everything immediately.

I also think it's OK for an implementation to do so: we have to be able to load it using `loadLibrary()` if it hasn't been done, and we have to be able to call `loadLibrary()` and succeed even in the situation where the library has been loaded, but nobody _guarantees_, at any point in time, that the library hasn't already been loaded.

I think it would be fine to test the situation in a more flexible way: It might complete normally, or it might throw.

I don't know if there is a variant of `Expect.throws` that allows for this kind of "multiple-choice outcomes", but otherwise it could be achieved using a regular `try`/`catch` statement.

Reply:
Never heard about this. Thank you, updated.